stdlib/public/Concurrency/AsyncStream.swift

```diff
@@ -110,25 +183,26 @@ public struct AsyncStream<Element> {
     }
 
     /// Resume the task awaiting the next iteration point by having it return
-    /// nil which signifies the end of the iteration.
+    /// nil, which signifies the end of the iteration.
     ///
-    /// Calling this function more than once is idempotent; i.e. finishing more
-    /// than once does not alter the state beyond the requirements of
-    /// AsyncSequence; which claims that all values past a terminal state are
-    /// nil.
+    /// Calling this function more than once has no effect. Once you call
+    /// finish, the stream enters a terminal state and produces no further
+    /// elements.
     public func finish() {
       storage.finish()
     }
 
-    /// A callback to invoke when iteration of a AsyncStream is cancelled.
+    /// A callback to invoke when canceling iteration of an asynchronous
+    /// stream.
     ///
-    /// If an `onTermination` callback is set, when iteration of a AsyncStream is
-    /// cancelled via task cancellation that callback is invoked. The callback
-    /// is disposed of after any terminal state is reached.
+    /// If an `onTermination` callback is set, using task cancellation to
+    /// terminate iteration of a `AsyncStream` results in a call to this
+    /// callback.
     ///
-    /// Cancelling an active iteration will first invoke the onTermination 
-    /// callback and then resume yeilding nil. This means that any cleanup state
-    /// can be emitted accordingly in the cancellation handler.
+    /// Cancelling an active iteration invokes the `onTermination` callback
+    /// first, then resumes by yielding `nil`. This means that you can perform
+    /// needed cleanup in the cancellation handler. After reaching a terminal
+    /// state, the `AsyncStream` disposes of the callback.
```

Comment:
I'm not exactly sure what "terminal state" means here. Probably something like "After the stream terminates..."  Likewise, what does it mean to "dispose of the callback"?  Is there a special action that developers need to take here, unlike other places where they provide a closure?

Reply:
> Is there a special action that developers need to take here, unlike other places where they provide a closure?

Could probably use a little guidance on the semantics from @phausler.